lib/ansible/modules/cloud/amazon/rds_snapshot.py

```diff
@@ -0,0 +1,196 @@
+    snapshot: new-database-snapshot
+    state: absent
+
+- debug:
+    msg: "The new db endpoint is {{ rds.instance.endpoint }}"
+'''
+
+RETURN = '''
+snapshot:
+  description: the information returned in data from boto3 get_db_snapshot
+  returned: success
+  type: dict
+'''
+
+import botocore
+import time
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn
+from ansible.module_utils.aws.rds import RDSSnapshot, get_db_snapshot
+
+# FIXME: the command stuff needs a 'state' like alias to make things consistent -- MPD
+
+
+def await_resource(conn, resource, status, module):
+    wait_timeout = module.params.get('wait_timeout') + time.time()
+    while wait_timeout > time.time() and resource.status != status:
+        time.sleep(5)
+        if wait_timeout <= time.time():
+            module.fail_json(msg="Timeout waiting for RDS resource %s" % resource.name)
+        # Temporary until all the rds2 commands have their responses parsed
+        if resource.name is None:
+            module.fail_json(
+                msg="There was a problem waiting for RDS snapshot %s" %
+                resource.snapshot)
+        resource = get_db_snapshot(conn, resource.name)
+    return resource
+
+
+def delete_snapshot(module, conn):
+    snapshot = module.params.get('snapshot')
+
+    result = get_db_snapshot(conn, snapshot)
+    if not result:
+        module.exit_json(changed=False)
+    if result.status == 'deleting':
+        module.exit_json(changed=False)
+    try:
+        result = conn.delete_db_snapshot(DBSnapshotIdentifier=snapshot)
+    except Exception as e:
+        module.fail_json(msg="Failed to delete snapshot: %s" % e.message)
+
+    # If we're not waiting for a delete to complete then we're all done
+    # so just return
+    if not module.params.get('wait'):
+        module.exit_json(changed=True)
+    try:
+        await_resource(conn, result, 'deleted', module)
+        module.exit_json(changed=True)
+    except Exception as e:
+        if e.response['Error']['Code'] == 'DBSnapshotNotFound':
+            module.exit_json(changed=True)
+        else:
+            module.fail_json(msg=e.message)
+
+
+def create_snapshot(module, conn):
+    instance_name = module.params.get('instance_name')
+    if not instance_name:
+        module.fail_json(msg='instance_name is required for rds_snapshot when state=present')
+    snapshot_name = module.params.get('snapshot')
+    changed = False
+    snapshot = get_db_snapshot(conn, snapshot_name)
+    if not snapshot:
+        try:
+            resource = conn.create_db_snapshot(DBSnapshotIdentifier=snapshot_name,
+                                               DBInstanceIdentifier=instance_name)
+            snapshot = RDSSnapshot(resource['DBSnapshot'])
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=e.message)
```

Comment:
See https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2
Can update most of the exception handling. Helpful for future debugging! :)